src/libc/allocator.src

```diff
@@ -4,6 +4,49 @@
 	public	_malloc, _free, _realloc
 
 	public	_calloc
+
+if defined __TICE__
+
+; uses the hardware specific $E40000 memory location
+
+; void *calloc(size_t nmemb, size_t size)
+_calloc:
+	pop	de
+	pop	bc
+	ex	(sp), hl
+	push	bc
+	push	de
+	call	__imulu
+	push	hl
+	push	hl
+	call	_malloc
+	pop	bc	; reset SP
+	; test for NULL
+	add	hl, bc
+	; or	a, a	; assumes that ptr + size does not overflow on TICE
+	sbc	hl, bc
+	pop	bc	; BC = size
```

Comment:
Revival of https://github.com/CE-Programming/toolchain/pull/631#discussion_r2328330994 including calc84's correction:

```suggestion
	pop	bc	; BC = size
	; test for NULL
	add	hl, bc
	; or	a, a	; assumes that ptr + size does not overflow
	sbc	hl, bc
```